specification/botservice/resource-manager/Microsoft.BotService/preview/2018-07-12/botservice.json

```diff
@@ -2666,10 +2881,37 @@
           },
           {
             "value": "SmsChannel"
+          },
+          {
+            "value": "LineChannel"
           }
         ]
       },
       "description": "The name of the Channel resource."
+    },
+    "regenerateKeyschannelNameParameter": {
+      "name": "channelName",
+      "x-ms-parameter-location": "method",
+      "in": "path",
+      "required": true,
+      "type": "string",
+      "enum": [
+        "WebChatChannel",
+        "DirectLineChannel"
+      ],
+      "x-ms-enum": {
+        "name": "channelName",
+        "modelAsString": false,
```

Comment:
These values will *never* change?  If there is a chance these values will change ina  future api-version, this should use modelAsString: true